ncm-network/src/main/pan/components/network/types/network/interface.pan

```diff
@@ -0,0 +1,166 @@
+    @{Rules for this interface.
+      These values are used to generate the /etc/sysconfig/network-scripts/rule[6]-<interface> files
+      as used by ifup-routes when using ncm-network.
+      This allows for mixed IPv4 and IPv6 configuration}
+    "rule" ? network_rule[]
+    @{Aliases for this interface.
+      These values are used to generate the /etc/sysconfig/network-scripts/ifcfg-<interface>:<key> files
+      as used by ifup-aliases when using ncm-network.}
+    "aliases" ? network_interface_alias{}
+    @{Explicitly set the MAC address. The MAC address is taken from /hardware/cards/nic/<interface>/hwaddr.}
+    "set_hwaddr" ? boolean
+
+
+    @{Is a VLAN device. If the device name starts with vlan, this is always true.}
+    "vlan" ? boolean
+    @{If the device name starts with vlan, this has to be set.
+      It is set (but ignored by ifup) if it the device is not named vlan}
+    "physdev" ? valid_interface
+
+    "fqdn" ? string
+    "network_environment" ? string
+    "network_type" ? string
+    "nmcontrolled" ? boolean
+    @{Set DEFROUTE, is the default for ipv6_defroute}
+    "defroute" ? boolean
+
+    "bridge" ? valid_interface
+    "linkdelay" ? long # LINKDELAY
+    "stp" ? boolean # enable/disable stp on bridge (true: STP=on)
+    "delay" ? long # brctl setfd DELAY
+    "bridging_opts" ? network_bridging_options
+
+    "bond_ifaces" ? string[]
+
+    "ipv4_failure_fatal" ? boolean
+    "ipv6_autoconf" ? boolean
+    "ipv6_failure_fatal" ? boolean
+    "ipv6_mtu" ? long(1280..65536)
+    "ipv6_privacy" ? string with match(SELF, '^rfc3041$')
+    "ipv6_rtr" ? boolean
+    @{Set IPV6_DEFROUTE, defaults to defroute value}
+    "ipv6_defroute" ? boolean
+    "ipv6addr" ? type_network_name
+    "ipv6addr_secondaries" ? type_network_name[]
+    "ipv6init" ? boolean
+
+    include network_interface_ethtool
+    include network_interface_ovs
+    include network_interface_tunnel
+} with {
+    network_interface_ovs_validate(SELF);
+
+    if ( exists(SELF['bond_ifaces']) ) {
+        foreach (i; iface; SELF['bond_ifaces']) {
+            if ( !exists("/system/network/interfaces/" + iface) ) {
+                error("The " + iface + " interface is used by bond_ifaces, but does not exist");
+            };
+        };
+    };
+    if (exists(SELF['ip']) && exists(SELF['netmask'])) {
+        if (exists(SELF['gateway']) && ! ip_in_network(SELF['gateway'], SELF['ip'], SELF['netmask'])) {
+            error('networkinterface has gateway %s not reachable from ip %s with netmask %s',
+                            SELF['gateway'], SELF['ip'], SELF['netmask']);
+        };
+        if (exists(SELF['broadcast']) && ! ip_in_network(SELF['broadcast'], SELF['ip'], SELF['netmask'])) {
+            error('networkinterface has broadcast %s not reachable from ip %s with netmask %s',
+                            SELF['broadcast'], SELF['ip'], SELF['netmask']);
+        };
+    };
+
+    network_interface_tunnel_validate(SELF);
```

Comment:
Would this make more sense at line 144? ie keep all the function calls together and then the block of code?

Reply:
i kept the order of tests, but sure, i can move this up